client/me/account/main.jsx

```diff
@@ -84,21 +82,50 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
+	},
 
-		valueLink.requestChange = ( value ) => {
-			const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+	updateUserSetting( settingName ) {
+		return event => this.props.userSettings.updateSetting( settingName, event.target.value );
+	},
 
-			this.props.userSettings.updateSetting( 'language', value );
-			if ( value !== originalLanguage ) {
-				this.setState( { redirect: '/me/account' } );
-			} else {
-				this.setState( { redirect: false } );
-			}
-		};
+	updateUserSettingCheckbox( settingName ) {
+		return event => this.props.userSettings.updateSetting( settingName, event.target.checked );
```

Comment:
This pattern further suggests to me that this generalisation shouldn't really be made.
`updateUserSetting` can't be used in a general way so we're having to add `updateUserSettingCheckbox` to compensate for checkbox's nuance.